### ISSUE TITLE
Enable staff with different numbers of lines 

### ISSUE THREAD

**Issue by reinierdevalk** [music-encoding/tablature-ig#23]
Staves can have fewer (or more?) than six lines.

Example:

`phalese-1547_7-no_21.png`
![phalese-1547_7-no_21](https://user-images.githubusercontent.com/14223345/114923379-09333580-9e2d-11eb-8c01-dfdc0be8b7b8.png)

**Comment by DILewis** [comment]
I think we have this already in our spec as `<staffdef lines="5"...` **but** I don't know if we can say that symbols go on lines or spaces (and which space)

**Comment by reinierdevalk** [comment]
@DILewis Ah, you're right. Should I change this issue's title into 'Enable placing symbols on or between staff lines'?

**Comment by lpugin** [comment]
If I understand correctly, this means that `@tab.course` in `note` can be smaller or greater than values indicated in `staffDef`. So we should allow `<note tab.course="0">` (or even `<note tab.course="-1">`?) for notes below the course / staff lines. Does it make sense?

**Comment by DILewis** [comment]
@lpugin Not quite. The semantics are constant here, it's just the placement of the symbols – some put put symbols on the lines as if the line is an instrumental course being stopped. Others put the symbol between lines so they're easier to read. If you've got 6 lines, you've got 7 spaces, so there's a choice whether to put the first course above or below the first line.
Does that make sense? Reinier probably has examples to hand.

**Comment by lpugin** [comment]
Alternatively, we could make some lines invisible. I think this could be indicated in the `staffDef/tuning`. Something like:
```xml
<staffDef lines="6">
  <tuning>
    <course n="1" visible="false"/>
  </tuning>
</staffDef>
```
However, setting visual parameter in tuning looks pretty wrong, so maybe this should go elsewhere in staffDef. I don't think we have in CNM a way to make one specific staff line invisible.

**Comment by reinierdevalk** [comment]
I think we should still use the actual course for those notes. Let's say we have a five-line staff, and a note on the sixth course (as in one of the examples). Then that note should still have `tab.course='6'`, but it might get a ledger staff line.

**Comment by lpugin** [comment]
OK. I was looking at that particular part:
![image](https://user-images.githubusercontent.com/689412/115008669-af258500-9eab-11eb-8676-6da5d863bdd9.png)

**Comment by reinierdevalk** [comment]
@lpugin Yes, that's how I understood you. David is talking about something else: whether the symobls are 'through' the lines (as on your example) or on top of them.

**Comment by DILewis** [comment]
Sorry, I wasn't thinking about the 5-line staff leger line issue when I replied. Not paying enough attention.

**Comment by reinierdevalk** [comment]
Should the _d_ on the sixth course in Laurent's example above be treated similar to a note on the 7th, 8th, etc. course, for which there will never be a staff line?

**Comment by reinierdevalk** [comment]
It looks like specifying in the StaffDef whether the symbols are rendered on top of or through the staff lines solves that issue, doesn't it? @lpugin an example of symbols on top of the staff line is this one:

![Screenshot 2021-04-16 at 12 20 31](https://user-images.githubusercontent.com/14223345/115010941-53103000-9eae-11eb-9edb-80b9dc159276.png)

**Comment by lpugin** [comment]
Actually `staffDef@ontheline` looks appropriate to make this distinction. For the latest example above I would assume expect:
```xml
<staffDef lines="5" ontheline="false"/>
```

**Comment by kepper** [comment]
well, `@ontheline` currently has a slightly different meaning (according to it's description). It `determines the placement of notes on a 1-line staff. A value of 'true' places all notes on the line, while a value of 'false' places stems-up notes above the line and stems-down notes below the line.` I'm not sure this default placement is sufficient for our use case, but we could model something very similar just for tab, maybe called `@ontheline.tab` (or some such).

**Comment by kepper** [comment]
Just to get back to the idea of invisible staff lines above, even though they seem to be less of a solution: There is `@lines`, but there's also `@lines.color` (see https://music-encoding.org/guidelines/dev/attribute-classes/att.staffDef.vis.html#attributes_class), which is supposed to get as many values as there are lines, and the documentation explicitly mentions that this may be used to make lines invisible…
